### PR TITLE
build: Update Apple platform conditionals

### PR DESCRIPTION
**Pull request by keith**
Previously this only checked if the desired platform was the "darwin"
CPU. To build for iOS we need to expand any conditionals for macOS to
apply to iOS as well (in the future they may differ more). This new
alias makes it easier to check the Apple CPUs we care about without
having to duplicate the check at each call site.

Risk Level: Low

### CODE REVIEW

**Comment by keith** [comment]
There's an issue with if you disable hot restarting, I'm debugging it. Also I intentionally didn't checkin the unrelated formatting changes but I will update with those.

**Comment by keith** [comment]
Actually it turns out the issue is an issue today as well. You can reproduce this by building on a Mac with this:

```
bazel build source/exe:envoy-static --define=hot_restart=disabled
```

You end up with this error:

```
ERROR: /Users/ksmiley/dev/envoy/source/exe/BUILD:26:1: Illegal ambiguous match on configurable attribute "copts" in //source/exe:envoy-static:
//bazel:disable_hot_restart
@bazel_tools//tools/osx:darwin
Multiple matches are not allowed unless one is unambiguously more specialized.
```

Which I don't see how to avoid besides by not passing the define on macOS.

File: bazel/BUILD [hunk 1]
```diff
@@ -164,3 +164,62 @@ alias(
...
+        {
+            ":darwin": ":darwin",
+            ":darwin_x86_64": ":darwin_x86_64",
+            ":ios_arm64": ":ios_arm64",
+            ":ios_arm64e": ":ios_arm64e",
+            ":ios_armv7": ":ios_armv7",
+            ":ios_armv7s": ":ios_armv7s",
+            ":ios_i386": ":ios_i386",
+            ":ios_x86_64": ":ios_x86_64",
+            "//conditions:default": ":darwin",  # This is unreachable
```
Comment (lizan) [inline bazel/BUILD]: if this is unreachable, use no_match_error?
https://docs.bazel.build/versions/master/configurable-attributes.html#custom-error-messages

Comment (keith) [reply bazel/BUILD]: This is pretty weird, which is why I linked the docs at the top which explain this further. This isn't _actually_ unreachable, _but_ since you return `:darwin` (or any other value in the dictionary here) and because if that was the current value, it would have been matched above, we know this codepath won't get chosen from the `select()` statement that calls this.

Comment (htuch) [reply bazel/BUILD]: I'm a bit confused by this explanation, can you rephrase?

Comment (keith) [reply bazel/BUILD]: Definitely! So you can think of this as a nested select statement where this one, the inner one, returns something we know will match (the equivalent of returning true) for apple conditions. So here if `ios_armv7s` is matched, we return the same value so that the parent select statement correctly matches that case. So it's possible that none of these statements in this nested select will be true, so we need the equivalent of returning false for "none of these matched" and since select doesn't have a structure for that, we return a value we know won't be match (because if it was matched we would have returned already before)

Comment (htuch) [reply bazel/BUILD]: Thanks for explaining. I would change the comment to `# If we are on non-Apple we will fall-thru to this default. By making the value :darwin, this is guaranteed to be False and any select statements referencing //bazel:apple will not match.`. Alternatively, we could add:
```
config_setting(
  name = "false",
  values = {"_never_match": "ignore"}
)
```

or something. I bet Bazel has an in-built you can use for this somewhere, might not be documented.

Comment (keith) [reply bazel/BUILD]: I don't see anything in src/conditions for this, but yea if you'd prefer that that's fine with me

Comment (keith) [reply bazel/BUILD]: Updated to do that, let me know what you think!

Comment (htuch) [reply bazel/BUILD]: Looks good. Needs CI fix. @lizan WDYT?

Comment (lizan) [reply bazel/BUILD]: Seems CI is not happy with that?

Comment (keith) [reply bazel/BUILD]: I updated to go back to the original returning `:darwin` and clarified that further in the comment. Turns out the values of the config settings are validated

**Comment by keith** [comment]
Thanks!
